### PR TITLE
Enable caching of catkin_pkg.packages.find_packages when using roslaunch

### PR DESCRIPTION
**Pull request by garaemon**
patch for https://github.com/ros/ros_comm/issues/541

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please consider to update the PR according to the comment made on the related PR https://github.com/ros/catkin/pull/704#issuecomment-68476827

File: tools/roslaunch/src/roslaunch/__init__.py [hunk 1]
```diff
@@ -203,7 +204,13 @@ def main(argv=sys.argv):
     try:
         from . import rlutil
         parser = _get_optparse()
-        
+        global catkin_packages_cache
+        catkin_packages_cache = roslib.packages.catkin_packages_cache()
+        # set cache for each modules
+        rlutil._catkin_packages_cache = catkin_packages_cache
```
Comment (dirk-thomas) [inline tools/roslaunch/src/roslaunch/__init__.py:210]: Accessing private variables is not recommended.

File: tools/roslaunch/src/roslaunch/node_args.py [hunk 2]
```diff
@@ -56,6 +56,8 @@
 from roslaunch.config import load_config_default
 import roslaunch.xmlloader
 
+_catkin_packages_cache = None
```
Comment (dirk-thomas) [inline tools/roslaunch/src/roslaunch/node_args.py:59]: Why do multiple files have this variable?

**Comment by wkentaro** [comment]
@dirk-thomas 
I sent PR with another implementation, so please review:
- https://github.com/ros/ros_comm/pull/676

And it depends:
- https://github.com/ros/ros/pull/91
- https://github.com/ros/catkin/pull/757

**Pull request closed without merging**